### DISCUSSION TITLE
Snapshots on Xcode Cloud with different line height than locally

### DISCUSSION THREAD

**Discussion by fespinoza** [pointfreeco/swift-snapshot-testing#842] [Q&A] [answered]
Hi

I managed to run snapshot test on Xcode Cloud (I'm preparing a YouTube video to explain my solution).

I have a sample repo project in https://github.com/fespinoza/sample-json-app

In https://github.com/fespinoza/sample-json-app/pull/6 I set up Xcode Cloud to be able to run snapshots.

Snapshot Context:
----------

- Development
    - I develop in a M1 Max machine
    - Xcode 15.3
    - iOS 17.4 - iPhone 15 Pro simulator
- CI
<details><summary>CI workflow setup</summary>
<p>
<img width="1003" alt="Screenshot 2024-03-26 at 20 59 58" src="https://github.com/pointfreeco/swift-snapshot-testing/assets/167989/61757c86-1e74-40ee-8140-9fa276d9ca87">

<img width="987" alt="Screenshot 2024-03-26 at 21 00 08" src="https://github.com/pointfreeco/swift-snapshot-testing/assets/167989/9fa221b2-23ce-4d7d-b1cb-f4d5f4e90d8e">
</p>
</details> 

- Snapshot details
    - `precision = 0.98`
    - `perceptualPrecision = 0.98`
- snapshot testing version `1.15.2` 

I use defined traits when taking snapshots

https://github.com/fespinoza/sample-json-app/blob/84b81ee3fa1d24e97bbeded65487b32258a42325/SampleProjectTests/Extensions/SnapshotTesting/XCTestCase-Extensions.swift#L100-L123

https://github.com/fespinoza/sample-json-app/blob/84b81ee3fa1d24e97bbeded65487b32258a42325/SampleProjectTests/Extensions/SnapshotTesting/SnapshotTestContext.swift#L75-L87

and I even attempt setting the `lineSpacing` manually

https://github.com/fespinoza/sample-json-app/blob/84b81ee3fa1d24e97bbeded65487b32258a42325/SampleProjectTests/MovieDetails/MovieDetailsViewTests.swift#L36

The problem:
---------

with all that setup the problem is

> [!WARNING]  
> **the line height of the reference is different from the one on CI**

I don't understand how this happens.

Any ideas?

| Reference | Failure |
|:----:|:-----:|
| ![reference](https://github.com/pointfreeco/swift-snapshot-testing/assets/167989/7a2a502b-7c86-4997-9d5d-8562a6c7aca1) | ![failure](https://github.com/pointfreeco/swift-snapshot-testing/assets/167989/d69b3b2f-6c7a-4b91-9208-67452ff14863) |

**Comment by meherkasam** [comment]
Did you compare the supported languages between your local simulator and CI simulator? The line height increases when you add a language like Hindi or Arabic as a supported language.

**Reply by fespinoza** [reply, marked as answer]
Interesting!

this comment reminded me that I hadn't set a test plan for the project. **When I added the test plan and configure the language to English and region to Norway (as my machine) that made it work**.

Thanks for the help!

**Reply by mushthak** [reply]
Interesting!

I was getting this weird behaviour. 
For me, in simulator, under the "Language and Region" settings, English and Hindi was listed as "PREFERRED LANGUAGES". 
Even though current language was set as English, the test was failing.  
When deleted Hindi from this list, the test started passing 🙂

<img src="https://github.com/user-attachments/assets/7de0cde1-344c-406e-8aca-6541d6a20901" width="100" height="200">
<img src="https://github.com/user-attachments/assets/0efc6c45-7841-4fdb-993b-afcc7262b1bb" width="100" height="200">

**Reply by baphony** [reply]
> Interesting!
> 
> this comment reminded me that I hadn't set a test plan for the project. **When I added the test plan and configure the language to English and region to Norway (as my machine) that made it work**.
> 
> Thanks for the help!

Many thanks for your answer @fespinoza!
I have spent so much time trying to figure out the reason, I first thought it was some wrong device layout / insets values, but setting the test plan language fixed the issue 👍
